Fix python and unit tests

ptschip opened the pull request.

ptschip committed:
Fix the may152018_forkactivation_2.py

Since the fork time has passed we need to explicity set the pre
fork time when we mined the first set of blocks.

ptschip committed:
Fix excessive.py so that it works after the may15th fork

ptschip committed:
Set the minining max block size in excessiveblock_test.cpp

This makes sure we're starting at the right miningsize in case
it was somehow set to a different value in a previous test.

sickpig commented on src/main.cpp:
```diff
@@ -572,7 +572,7 @@ bool IsMay152018Next(const Consensus::Params &consensusparams, const CBlockIndex
         return false;
     }
-
+printf("mining for value %ld\n", miningForkTime.value);
```
this will probably make travis fail due to formatting checker (no big deal since is temporary)

sickpig commented on src/policy/policy.cpp:
```diff
@@ -40,6 +40,7 @@ bool IsStandard(const CScript &scriptPubKey, txnouttype &whichType)
     if (!Solver(scriptPubKey, whichType, vSolutions))
         return false;
 
+printf("scriptpubkeysiz %d nmaxdatacarrie size %d\n", scriptPubKey.size(), nMaxDatacarrierBytes);
```
same here

sickpig commented on src/test/transaction_tests.cpp:
```diff
@@ -368,12 +368,14 @@ BOOST_AUTO_TEST_CASE(test_IsStandard)
         CScript() << OP_RETURN
                   << ParseHex("04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef3804678afd"
-                              "b0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef3800");
+                              "b0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef3838");
```
just curios why changing the last 2 digits?

sickpig commented on src/main.cpp:
```diff
@@ -2567,7 +2567,7 @@ bool DisconnectTip(CValidationState &state, const Consensus::Params &consensusPa
     // If this block enabled the may152018 opcodes, then we need to
     // clear the mempool of any transaction using them.
-    if (IsMay152018Enabled(consensusParams, pindexDelete) && !IsMay152018Enabled(consensusParams, pindexDelete->pprev))
+    if (IsMay152018Next(consensusParams, chainActive.Tip()))
```
Aren't these 2 conditions equivalent? 

`pindexDelete` is initialized to chainActiveTip() 

`IsMay152018Enabled` return `true` if current height belongs to `[x-1,+inf)` where x is the first block mined after MTP conditions triggered.

So the conditions that is changed in this PR will return true only if we are in `[x-1,x-1]` which is the same thing that  `IsMay152018Next` cause it's based `forkAtNextBlock`

ptschip replied:
you know what , you're right but fact is , it doesn't work...but now i look at it again, i think there is something wrong with pindexDelete->pprev, because if i use it in IsMay152018Next(pindexDelete->pprev) then the script fails.

ptschip committed:
fix transaction_tests.cpp

assigne the value of 83 to maxdatacarrierbytes in case it was changed
prior to running this set of tests.

ptschip committed:
fix mempool_reorg.py to work after fork activation.

ptschip committed:
small fix for mempool_reorg
